manifests/charts/istio-control/istio-discovery/templates/deployment.yaml

```diff
@@ -169,9 +169,10 @@ spec:
           # Also, check for an explicit ENV override (legacy approach) and prefer that
           # if present
           {{ $ztTrustedNS := or .Values.trustedZtunnelNamespace .Release.Namespace }}
+          {{ $ztTrustedName := or .Values.trustedZtunnelName "ztunnel" }}
```

Comment:
@howardjohn  I know this has been merged already, but anyway: the istiod and ztunnel charts are not in sync here. In version 1.25.0, the ztunnel chart was broken because it created an `istio` service account but hard-coded the ztunnel DaemonSet ("istio") to use `ztunnel`. In version 1.25.1, this was fixed so that `istio` is used as the service account in the DaemonSet. Having `ztunnel` as the default in the istiod chart is not compatible with the ztunnel chart, it breaks the generation of mTLS certificates.

Reply:
You can set `trustedZtunnelName` if you want to override ztunnel to be a different name. Though its easiest if you just keep it `ztunnel`

Reply:
@howardjohn Actually, on further investigation, the problem (for me) is that the ztunnel chart is using the helm release name as the default name for the resources it creates. This design choice is a little weird. Since I install all the istio charts as sub-charts of a main "istio" chart, and the helm release is also "istio", it meant that the ztunnel chart was creating a DaemonSet and ServiceAccount named "istio", which doesn't match the default in istiod. Fortunately, this behavior can be overridden by setting `ztunnel.resourceName: ztunnel`. A little awkward, but it works.